ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/ClientIntegrationTest.java

```diff
@@ -708,31 +707,38 @@ public void shouldListTopics() throws Exception {
     final List<TopicInfo> topics = client.listTopics().get();
 
     // Then
-    assertThat(topics, containsInAnyOrder(
+    assertThat("" + topics, topics, containsInAnyOrder(
         topicInfo(TEST_TOPIC),
         topicInfo(EMPTY_TEST_TOPIC),
         topicInfo(EMPTY_TEST_TOPIC_2),
         topicInfo(AGG_TABLE)
     ));
   }
 
-  @Ignore // temporarily disable while flakiness is being investigated
   @Test
-  public void shouldListQueries() throws Exception {
+  public void shouldListQueries() {
     // When
-    final List<QueryInfo> queries = client.listQueries().get();
+    // Try multiple times to allow time for queries started by the other tests to finish terminating
```

Comment:
I wonder if this can affect stability of any of the other tests - is there a way to assert this in the `@Before` or `@After` methods?

Reply:
I considered that but doing so would mean one failing test could cause all other tests to fail. For example, if a test starts a persistent query and fails before it's able to terminate it, then all subsequent tests will fail in the `@Before` or `@After` which doesn't seem great.

I also considered adding this assertion to the end of each test that starts a transient query, so that those tests wait for the transient query to be fully terminated before the test completes. However this isn't great either because the list queries test still needs this retry loop, in case someone adds a new test that starts a transient query and forgets to add the assertion block to the end.

In light of that, I thought it made the most sense to have tests that require a certain number of queries assert that at the start of the test. WDYT?

Reply:
that's fair enough 👍